File: dataset/avatar/Python/Code/codeforces_282_B.py
firstline = int ( input ( ) )
total = 0
memory = [ ]
for x in range ( firstline ) :
    A , G = list ( map ( int , input ( ).split ( ) ) )
    if total + A <= 500 :
        total += A
        memory.append ( "A" )
    else :
        total -= G
        memory.append ( "G" )
print ( "".join ( memory ) )

File: output/gpt_4o/avatar/algo_based_translation/python/java/successful_translation/codeforces_282_B.java

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class codeforces_282_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int firstline = Integer.parseInt(scanner.nextLine());
        int total = 0;
        List<String> memory = new ArrayList<>();
        
        for (int x = 0; x < firstline; x++) {
            String[] inputs = scanner.nextLine().split(" ");
            int A = Integer.parseInt(inputs[0]);
            int G = Integer.parseInt(inputs[1]);
            
            if (total + A <= 500) {
                total += A;
                memory.add("A");
            } else {
                total -= G;
                memory.add("G");
            }
        }
        
        System.out.println(String.join("", memory));
        
        scanner.close();
    }
}

